src/Avalonia.Controls/TextBox.cs

```diff
@@ -1470,7 +1459,7 @@ protected override void OnPointerPressed(PointerPressedEventArgs e)
             var text = Text;
             var clickInfo = e.GetCurrentPoint(this);
 
-            if (text != null && clickInfo.Properties.IsLeftButtonPressed &&
+            if (text != null && (e.Pointer.Type == PointerType.Mouse || e.ClickCount >= 2) && clickInfo.Properties.IsLeftButtonPressed &&
```

Comment:
I'm not sure what the new condition does, but it seems incompatible with the code inside the `if` branch: `isTouch && e.ClickCount == 1` below can now never be `true`

Reply:
Unresolving this as dragging the caret itself doesn't work (see my comment below).